lib/client/keystore.go

```diff
@@ -582,10 +582,10 @@ func (fs *fsLocalNonSessionKeyStore) kubeCertPath(idx KeyIndex, kubename string)
 	return keypaths.KubeCertPath(fs.KeyDir, idx.ProxyHost, idx.Username, idx.ClusterName, kubename)
 }
 
-// acquireFileLock is trying to lock the file, until it's successful or timeout is exceeded.
-// File will be created if it doesn't exist.
+// acquireFileLock is trying to acquire lock, until it's successful or timeout is exceeded.
+// Lock file for the path will be created if it doesn't exist.
 func acquireFileLock(filePath string, timeout time.Duration) (func() error, error) {
-	fileLock := flock.New(filePath)
+	fileLock := flock.New(filePath + ".lock")
```

Comment:
Do we want to lock it "properly" on Unix systems (i.e. lock the actual file) and have a different implementation for Windows? The benefit would be that the file would be protected by external modifications also (on Unix systems at least).

Reply:
It doesn't really provide protection from external modification for *nix, because it's advisory locking. It works only as long as process is cooperating, but some non-cooperating process can still modify the file. Essentially it's like a mutex - as long as code tries to lock first, everything's good, but nothing really stops direct modification of protected variable :)
But fortunately I don't think we'll have a problem here, since only we are writing to `known_hosts`.

Reply:
Will this approach leave .lock files behind? Might be confusing unless .lock file is also deleted when the lock is released.

Reply:
Yes, currently .lock files are left behind, because their presence doesn't matter to the actual locking. But I can add removal on unlocking for regular operations. It will be left over in case process dies unexpectedly, but as I said before, it will not matter and it will be just deleted on next unlocking.

Reply:
Having a .lock file always present may be confusing for users, I think it would be preferable if we didn't leave it behind.

Reply:
Refactored `utils/fs` as we discussed.